### DISCUSSION TITLE
不同域名自定义顶栏的搜索历史不互通

### DISCUSSION THREAD

**Discussion by Jyanr** [the1812/Bilibili-Evolved#3893] [功能建议 / Ideas]
[www](https://www.bilibili.com/) 和 [search](https://search.bilibili.com/)
两个域名下的搜索历史是完全不一样的
并需要分开清除

希望合并一下

**Comment by the1812** [comment]
合并不了, 都不是同一个域名, b 站原生的搜索也是不互通的

**Reply by Jyanr** [reply]
> 合并不了, 都不是同一个域名, b 站原生的搜索也是不互通的

原生是可以的

这是原生搜索页的搜索历史，红圈是首页的搜索历史，用组件是没有的，只有蓝圈的

![屏幕截图_20221229_080001](https://user-images.githubusercontent.com/88956096/209887089-eb9a8a45-6d8b-4739-b978-c98681ae7abd.png)

**Reply by the1812** [reply]
原来已经改版了, 是我落后了

**Reply by pbxs666** [reply]
> 原来已经改版了, 是我落后了

搜索历史跨域有2个bug：
1.实际生效的并不是合并功能，更像是覆盖功能。我是在 www.bilibili.com/account/history **历史记录**页面下更新2.6.1预览版的。更新后该域名（www.bilibili.com和www.bilibili.com/……）的搜索历史保留，随便搜一个词，发现 search.bilibili.com和search.bilibili.com/…… **搜索页面**的搜索记录被覆盖了，同样的，space.bilibili.com**个人空间**也被覆盖了；
2.现在删除单条历史记录的功能失效，点击历史词条后面的叉号无效，只有**清除搜索历史**功能有效，关键是全部清除这个功能我基本用不到啊，只想删单条记录。
另外新发现，search.bilibili.com这个页面（仅仅这个页面，不是指域名），页面的原生搜索框里的单个搜索历史词条可以删除，并且也会同步到**自定义顶栏**的搜索历史，目前只能按照此法删单个搜索历史了。
![QQ截图20230209105416](https://user-images.githubusercontent.com/47630168/217706500-d6c14679-c016-441f-bc21-b7be45f40361.jpg)
